WMF Framework/Significant Events Endpoint/SignificantEventsModels.swift

```diff
@@ -0,0 +1,616 @@
+            public let title: String
+            public let lastName: String?
+            public let firstName: String?
+            public let yearPublished: String?
+            public let locationPublished: String?
+            public let publisher: String?
+            public let pagesCited: String?
+            public let isbn: String?
+            
+            init?(dict: [String: String]) {
+                guard let title = dict.nonEmptyValueForKey(key: "title") else {
+                    return nil
+                }
+                
+                self.title = title
+                
+                let batch1 = dict.nonEmptyValueForKey(key: "last") ??
+                    dict.nonEmptyValueForKey(key: "last1") ??
+                    dict.nonEmptyValueForKey(key: "author") ??
+                    dict.nonEmptyValueForKey(key: "author1") ??
+                    dict.nonEmptyValueForKey(key: "author1-last")
+                let batch2 = dict.nonEmptyValueForKey(key: "author-last") ??
+                    dict.nonEmptyValueForKey(key: "surname1") ??
+                    dict.nonEmptyValueForKey(key: "author-last1") ??
+                    dict.nonEmptyValueForKey(key: "subject1") ??
+                    dict.nonEmptyValueForKey(key: "surname")
+                let batch3 = dict.nonEmptyValueForKey(key: "author-last") ??
+                    dict.nonEmptyValueForKey(key: "subject")
+                
+                self.lastName = batch1 ?? batch2 ?? batch3
+                
+                self.firstName = dict.nonEmptyValueForKey(key: "first") ??
+                                dict.nonEmptyValueForKey(key: "given") ??
+                                dict.nonEmptyValueForKey(key: "author-first") ??
+                                dict.nonEmptyValueForKey(key: "first1") ??
+                                dict.nonEmptyValueForKey(key: "given1") ??
+                                dict.nonEmptyValueForKey(key: "author-first1") ??
+                                dict.nonEmptyValueForKey(key: "author1-first")
+                
+                self.yearPublished = dict.nonEmptyValueForKey(key: "year")
+                self.locationPublished = dict.nonEmptyValueForKey(key: "location") ??
+                                            dict.nonEmptyValueForKey(key: "place")
+                
+                self.publisher = dict.nonEmptyValueForKey(key: "publisher") ??
+                                dict.nonEmptyValueForKey(key: "distributor") ??
+                                dict.nonEmptyValueForKey(key: "institution") ??
+                                dict.nonEmptyValueForKey(key: "newsgroup")
+                
+                self.pagesCited = dict.nonEmptyValueForKey(key: "pages") ??
+                    dict.nonEmptyValueForKey(key: "pp")
+                
+                self.isbn = dict.nonEmptyValueForKey(key: "isbn") ??
+                            dict.nonEmptyValueForKey(key: "ISBN13") ??
```

Comment:
This is another opportunity where case insensitive `String` operations might reduce code. You could probably even adapt your `Dictionary` helper to support that pretty easily - maybe something like:

```swift

dict.nonEmptyValueForKey(key: "isbn13", caseInsensitive: true)

...

private extension Dictionary where Key == String, Value == String {
    func nonEmptyValueForKey(key: String, caseInsensitive: Bool = false) -> String? {
        guard let key = caseInsensitive 
                ? keys.first(where: {$0.caseInsensitiveCompare(key) == .orderedSame})
                : key else { 
                    return nil		
         }

        if let value = self[key], !value.isEmpty {
            return value
        }

        return nil
    }
}
```

Reply:
Good catch, this is done.